fsnotify.go

```diff
@@ -63,8 +73,33 @@ func (e Event) String() string {
 	return fmt.Sprintf("%q: %s", e.Name, e.Op.String())
 }
 
-// Common errors that can be reported by a watcher
-var (
-	ErrNonExistentWatch = errors.New("can't remove non-existent watcher")
-	ErrEventOverflow    = errors.New("fsnotify queue overflow")
-)
+// findDirs finds all directories under path (return value *includes* path as
+// the first entry).
+func findDirs(path string) ([]string, error) {
+	dirs := make([]string, 0, 8)
```

Comment:
What's the origin of the capacity value here?

Reply:
It's basically a micro-optimisation; the performance diffence between allocting an empty slice and a slice with cap=8 is negible, but if we start from 0 then adding 8 files will require a bunch of allocations for new backing arrays + copies (it doubles every time: so 1, 2, 4, 8), which is much more expensive, comparatively speaking:

	BenchmarkAllocEmpty-4             1000000000         0.2598 ns/op      0 B/op     0 allocs/op
	BenchmarkAllocEight-4             1000000000         0.2592 ns/op      0 B/op     0 allocs/op
	BenchmarkFillEightFromEmpty-4        7502155     157.9 ns/op         112 B/op     3 allocs/op
	BenchmarkFillEightFromAlloc-4      226564515         5.270 ns/op       0 B/op     0 allocs/op

I mean, ~155ns isn't all that much, but it's just as easy to type one over the other.

Why 8 and not 4 or 16? I felt 8 was a good trade-off between potential "wasted" memory and run-time speed.